### ISSUE TITLE
Handling the http response when the body is null or empty throws an exception

### ISSUE THREAD

**Issue by eugenpodaru** [skyleague/therefore#187]
### Description

When generating an http client that uses ky, there is an edge case that it not properly handled. When the response body is empty/null (in case of status code 204) and the expected response type is 'json' there are 2 ways of handing the response as shown bellow. 
The 1 pattern results in an error being thrown: Unexpected end of JSON input
The 2 pattern results in the _body var being an empty string, which is the expected outcome when dealing with such response bodies.

Therefore however opts for pattern 1, so the generated client throws an exception.

#### 1
```
nock('https://localhost/mambu').post('/loans').once().reply(204, undefined)

const client = ky.extend({ prefixUrl: 'https://localhost/mambu', throwHttpErrors: false })
const result = await client.post('loans')
const _body = await result.json()
```

#### 2
```
nock('https://localhost/mambu').post('/loans').once().reply(204, undefined)

const client = ky.extend({ prefixUrl: 'https://localhost/mambu', throwHttpErrors: false })
const _body = await client.post('loans').json()
```

**Comment by pevisscher** [comment]
The current version of nock does not support native fetch, are you using the beta version they provided that is going to support that? (and if not could it be that nock is not behaving as intended in the first place)

I will try to narrow this down further, also weird that we didnt see this happen with got

**Comment by eugenpodaru** [comment]
@pevisscher yes, I am using the beta version of nock. I can see this in real life, not only using nock.

**Comment by eugenpodaru** [comment]
@pevisscher you can see [here](https://github.com/sindresorhus/ky/blob/585ebcb80545d784b31033e5a70326a0eb202468/source/core/Ky.ts#L81-L107) why it behaves like that. When calling directly json() without awaiting, it will hit the shortcut they implemented there for dealing with this case. While is you call json() after awaiting, it will call the method on the actual response from undici, which probably does not handle this case.

**Comment by pevisscher** [comment]
Fixed by #189

**Comment by eugenpodaru** [comment]
@pevisscher the fix only partially fixes the issue. In case of other http status codes like 200 and 202 that return nothing (content-length=0) it will also fail.

**Comment by eugenpodaru** [comment]
the way I fixed it was in the awaitResponse method from the client to do the changes bellow. Do you see any drawback in doing that?

from
```
const result = await response
const _body = (await (responseType !== undefined ? result[responseType]() : result.text())) as I
```

to
```
const result = await response
const _body = (await (responseType !== undefined ? response[responseType]() : response.text())) as I
```

**Comment by pevisscher** [comment]
You're faster than me, just pushed the rest of the fix :) #190
